minecraft/nbt/decode.go

```diff
@@ -397,6 +434,36 @@ func (d *Decoder) unmarshalTag(val reflect.Value, t tagType, tagName string) err
 			if vk := val.Kind(); vk == reflect.Interface && val.NumMethod() != 0 {
 				return InvalidTypeError{Off: d.r.off, FieldType: val.Type(), Field: tagName, TagType: t}
 			}
+
+			// Fast path for TAG_Compound into map[string]any and any(map[string]any).
+			// Avoid reflect.Map SetMapIndex; reuse+clear maps and fill via native assignments.
+			if val.Kind() == reflect.Map && val.Type() == mapStringAnyType {
+				m := val.Interface().(map[string]any)
+				if m == nil {
+					m = make(map[string]any)
+					val.Set(reflect.ValueOf(m))
+				} else {
+					clear(m)
+				}
+				if err := d.unmarshalCompoundAny(m); err != nil {
+					return err
+				}
+				break
+			}
+			if val.Kind() == reflect.Interface && isAny(val) {
+				m, _ := val.Interface().(map[string]any)
+				if m == nil {
+					m = make(map[string]any)
+				} else {
+					clear(m)
+				}
+				val.Set(reflect.ValueOf(m))
+				if err := d.unmarshalCompoundAny(m); err != nil {
+					return err
+				}
+				break
```

Comment:
These two blocks are largely identical. Is there any way we could re-use the code?